x-pack/test/detection_engine_api_integration/security_and_spaces/group1/perform_bulk_action.ts

```diff
@@ -562,6 +562,46 @@ export default ({ getService }: FtrProviderContext): void => {
         expect(deleteIndexRule.index).to.eql(['initial-index-*', 'index2-*']);
       });
 
+      it('should add an index pattern to a rule and overwrite the data view', async () => {
+        const ruleId = 'ruleId';
+        const dataViewId = 'index1-*';
+        const simpleRule = {
+          ...getSimpleRule(ruleId),
+          indices: undefined,
+          data_view_id: dataViewId,
+        };
+        await createRule(supertest, log, simpleRule);
+
+        const { body: setIndexBody } = await postBulkAction()
+          .send({
+            query: '',
+            action: BulkAction.edit,
+            [BulkAction.edit]: [
+              {
+                type: BulkActionEditType.set_index_patterns,
+                value: ['initial-index-*'],
+                overwriteDataViews: true,
+              },
+            ],
+          })
+          .expect(200);
+
+        expect(setIndexBody.attributes.summary).to.eql({ failed: 0, succeeded: 1, total: 1 });
+
+        // Check that the updated rule is returned with the response
+        expect(setIndexBody.attributes.results.updated[0].index).to.eql(['initial-index-*']);
```

Comment:
should we also check if `data_view_id` has been set to undefined?

Reply:
yes definitely. I'll update this.